File: instautils/src/main/java/com/sanjaydevtech/instautils/InstaPost.java
package com.sanjaydevtech.instautils;

/**
 * InstaPost Object
 */
public class InstaPost {
    private String url;
    private int type;
    private String thumbnailUrl;
    public static final int INSTA_IMAGE = 0;
    public static final int INSTA_VIDEO = 1;

    InstaPost(String url, int type, String thumbnailUrl) {
        this.url = url;
        this.type = type;
        this.thumbnailUrl = thumbnailUrl;
    }

    /**
     * get the post download url
     *
     * @return a String
     */
    public String getUrl() {
        return url;
    }

    /**
     * get the type
     *
     * @return a int
     */
    public int getType() {
        return type;
    }

    /**
     * get the thumbnail
     *
     * @return a String
     */
    public String getThumbnailUrl() {
        return thumbnailUrl;
    }
}

File: instautils/src/main/java/com/sanjaydevtech/instautils/InstaPost.kt
package com.sanjaydevtech.instautils

/**
 * InstaPost Object
 *
 * [url] holds the actual image or video url
 *
 * [type] may be constants [INSTA_IMAGE] or [INSTA_VIDEO] or [INSTA_PROFILE]
 *
 * [thumbnailUrl] is the same as url for type [INSTA_IMAGE],
 * but a low res image for [INSTA_VIDEO] and [INSTA_PROFILE]
 */
data class InstaPost internal constructor(
        val url: String,
        val type: Int,
        val thumbnailUrl: String) {

    companion object {
        /** InstaPost object is from a Image */
        const val INSTA_IMAGE = 0

        /** InstaPost object is from a Video */
        const val INSTA_VIDEO = 1

        /** InstaPost object is from a Instagram Profile */
        const val INSTA_PROFILE = 2
    }
}